dev-tools/scripts/smokeTestRelease.py

```diff
@@ -1264,155 +1068,48 @@ def parse_config():
-      # Mixed version test case; ignore it for our purposes because we only
-      # tally up the "tests single Lucene version" indices
-      continue
-    elif name == '5.0.0.singlesegment':
-      tup = 5, 0, 0
-    else:
-      raise RuntimeError('could not parse version %s' % name)
-
-    testedIndices.add(tup)
-
-  l = list(testedIndices)
-  l.sort()
-  if False:
-    for release in l:
-      print('  %s' % '.'.join(str(x) for x in release))
-
-  allReleases = set(allReleases)
-
-  for x in testedIndices:
-    if x not in allReleases:
-      # Curious: we test 1.9.0 index but it's not in the releases (I think it was pulled because of nasty bug?)
-      if x != (1, 9, 0):
-        raise RuntimeError('tested version=%s but it was not released?' % '.'.join(str(y) for y in x))
-
-  notTested = []
-  for x in allReleases:
-    if x not in testedIndices:
-      releaseVersion = '.'.join(str(y) for y in x)
-      if releaseVersion in ('1.4.3', '1.9.1', '2.3.1', '2.3.2'):
-        # Exempt the dark ages indices
-        continue
-      if x >= tuple(int(y) for y in smokeVersion.split('.')):
-        # Exempt versions not less than the one being smoke tested
-        print('      Backcompat testing not required for release %s because it\'s not less than %s'
-              % (releaseVersion, smokeVersion))
-        continue
-      notTested.append(x)
-
-  if len(notTested) > 0:
-    notTested.sort()
-    print('Releases that don\'t seem to be tested:')
-    failed = True
-    for x in notTested:
-      print('  %s' % '.'.join(str(y) for y in x))
-    raise RuntimeError('some releases are not tested by TestBackwardsCompatibility?')
-  else:
-    print('    success!')
-
 
 def main():
   c = parse_config()
 
-  scriptVersion = scriptutil.find_current_version()
+  # Pick <major>.<minor> part of version and require script to be from same branch
+  scriptVersion = re.search(r'((\d+).(\d+)).(\d+)', scriptutil.find_current_version()).group(1).strip()
   if not c.version.startswith(scriptVersion + '.'):
     raise RuntimeError('smokeTestRelease.py for %s.X is incompatible with a %s release.' % (scriptVersion, c.version))
 
   print('NOTE: output encoding is %s' % sys.stdout.encoding)
   smokeTest(c.java, c.url, c.revision, c.version, c.tmp_dir, c.is_signed, c.local_keys, ' '.join(c.test_args),
             downloadOnly=c.download_only)
 
+
 def smokeTest(java, baseURL, gitRevision, version, tmpDir, isSigned, local_keys, testArgs, downloadOnly=False):
   startTime = datetime.datetime.now()
 
-  # disable flakey tests for smoke-tester runs:
-  testArgs = '-Dtests.badapples=false %s' % testArgs
-  
+  # Tests annotated @Nightly are more resource-intensive but often cover
+  # important code paths. They're disabled by default to preserve a good
+  # developer experience, but we enable them for smoke tests where we want good
+  # coverage. Still we disable @BadApple tests
+  testArgs = '-Dtests.nigthly=true -Dtests.badapples=false %s' % testArgs
```

Comment:
```suggestion
  testArgs = '-Dtests.nightly=true -Dtests.badapples=false %s' % testArgs
```

Reply:
Hawk eyes! 👁

Reply:
However, even if Lucene wants to run all nightly tests in smoke tester, are we sure that Solr wants to? I fear that several `@Nightly` tests are unstable, but I don't have too much facts to support it. They will for sure take a loong time and need some resources. I believe that before November 2021 the smoke tester has ran without nightly tests. Guess we'll just have to try. If they constantly fail or cause other issues, we can change it back to false.

Reply:
Great catch, I'll fix now.